src/sentry/lang/native/symbolicator.py

```diff
@@ -360,6 +361,18 @@ def _ensure_open(self):
         if not self.session:
             raise RuntimeError("Session not opened")
 
+    def _process_response(self, json):
+        source_names = {source["id"]: source.get("name") for source in self.sources}
+        source_names[INTERNAL_SOURCE_NAME] = "Sentry"
+
+        for module in json.get("modules") or ():
+            for candidate in module.get("candidates") or ():
+                candidate["source_name"] = (
+                    source_names.get(candidate["source"]) or "Unnamed Repository"
```

Comment:
The fallback of "Unnamed Repository" is aligned with the settings URL:

https://github.com/getsentry/sentry/blob/b8bf85948d4c4832c60ab31e9f1a5a3c39fd53cf/src%2Fsentry%2Fstatic%2Fsentry%2Fapp%2Fdata%2Fforms%2FprojectDebugFiles.jsx#L112

I'm realizing that this is something we should be doing in the UI, however.

Reply:
@priscilawebdev I think using the same `<em>{t('<Unnamed Repository>')}</em>` in the UI for candidates without a `source_name` would be the best option here.